lib/dotcom/trip_plan/itinerary_group.ex

```diff
@@ -0,0 +1,73 @@
+defmodule Dotcom.TripPlan.ItineraryGroup do
+  @moduledoc """
+  A single group of related itineraries
+  """
+
+  alias Dotcom.TripPlan.Itinerary
+
+  defstruct [:itineraries, :representative_index, :representative_time, :summary]
+
+  @type summarized_leg :: %{
+          routes: [Routes.Route.t()],
+          walk_minutes: non_neg_integer()
+        }
+
+  @type summary :: %{
+          accessible?: boolean() | nil,
+          duration: non_neg_integer(),
+          start: DateTime.t(),
+          stop: DateTime.t(),
```

Comment:
Nit (non-blocking): The names `start` and `stop` don't convey that these are times. And `stop` more frequently refers to a location where a bus allows passengers to embark and disembark. I would advocate for `start_time` and `end_time`.

(Note: This is only non-blocking because I know that the names aren't being introduced in this PR, and were like this already, and that fixing them could lead to PR bloat.)